CONTRIBUTING.md

```diff
@@ -1,95 +1,148 @@
-# Contributing to notary
+# Contributing Guidelines
 
-## Before reporting an issue...
+Welcome to the Notary Project community. The Notary Project accepts contributions via GitHub issues or pull requests. This document outlines the process to help get your contribution accepted. Please see the [Notary Project Overview](https://github.com/notaryproject/.github/blob/main/README.md) to learn about Notary Project.
 
-### If your problem is with...
+## Reporting Security Issues
 
- - automated builds
- - your account on the [Docker Hub](https://hub.docker.com/)
- - any other [Docker Hub](https://hub.docker.com/) issue
+> **Warning** If you are reporting a security vulnerability, Please DO NOT file a public issue.
 
-Then please do not report your issue here - you should instead report it to [https://support.docker.com](https://support.docker.com)
+Please refer to [Security Guide](SECURITY.md) on how to report security vulnerabilities.
 
-### If you...
+## Reporting Other Issues
 
- - need help setting up notary
- - can't figure out something
- - are not sure what's going on or what your problem is
+It is a great way to contribute to the Notary Project by reporting issues. Please open an issue on Github and follow the template to fill in required information.
 
-Then please do not open an issue here yet - you should first try one of the following support forums:
+Before opening any issue, please look up the existing issues to avoid submitting a duplication. If you found an issue that describes your problem:
 
- - irc: #docker-trust on freenode
+- Please read other user comments first to confirm if they have experienced the same issue. Keep in mind that a given error condition might be indicative of different problems. You may also find a workaround in the comments
+- To receive updates on an issue, simply click the 'subscribe' button
+- Please only comment if you have new, technical, and relevant information to add to the case.
 
-## Reporting an issue properly
+Issues will be marked with different labels for different purpose, see [Labels](#labels) for details.
 
-By following these simple rules you will get better and faster feedback on your issue.
+### Issue Lifecycle
 
- - search the bugtracker for an already reported issue
+The issue lifecycle is mainly driven by the maintainers.
 
-### If you found an issue that describes your problem:
+1. Issue creation:
+   - After an issue is created, it will be labelled `bug` or `enhancement`, and `triage`.
+2. Triage: The maintainers will triage the issues on a regular basis. The results of triage could be one of the following:
+   - Remove label `triage`, and add the issues to the correct milestone
+   - Remove label `triage`, and labelled `question` to indicate this issue need further information. After this issue is clarified with further information, it could be added to the correct milestone or labelled `wont fix`.
```

Comment:
```suggestion
   - Remove the 'triage' label and apply the 'question' label to indicate that this issue requires additional information. Once this issue has been clarified with the necessary details, it can then be assigned to the appropriate milestone or labeled as 'wont fix'.
```